### ISSUE TITLE
Increased CPU usage

### ISSUE THREAD

**Issue by kozerskil** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#71]
After updating to 1.0.5 CPU usage and temperature have increased. https://imgur.com/a/U43Z7nF With green line highlighted time where I restarted HA after updating the component. Hardware: RPi3 with HDD.

**Comment by PiotrMachowski** [comment]
Was this component working for you before update?

**Comment by kozerskil** [comment]
Well, I have it for quite some time, however there was this issue that after some time the map didn't refresh.

**Comment by PiotrMachowski** [comment]
So increased CPU usage is probably a difference between working and not working integration. Unfortunately this component have to consume a bit of CPU: it downloads and parses an image every 5 seconds. You can reduce it by changing scale to 1, trimming map and increasing scan interval.

**Comment by kozerskil** [comment]
Makes sense. I'll adjust the settings. I have one more idea tho: would it be possible to have attributes only (like vacuum_position) without the image?

**Comment by PiotrMachowski** [comment]
It is possible to do it for most of attributes, but not for all of them - some of them are calculated using map image

**Comment by kozerskil** [comment]
ok, I'd like to test, but how to make the component **not** draw anything?

**Comment by PiotrMachowski** [comment]
Oh, sorry, my answer was misleading. It is possible for me to implement it, it is not possible in current version of integration

**Comment by kozerskil** [comment]
Point taken, it's up to you then :)
